addons/godot_xml/xml.gd

```diff
@@ -128,12 +128,14 @@ static func _parse(xml: PackedByteArray) -> XMLDocument:
 		if node == null:
 			continue
 
+		var node_type = parser.get_node_type()
+		if node_type == XMLParser.NODE_ELEMENT_END and (queue.is_empty() || queue.back().name != node.name):
+			continue
+
```

Comment:
What do these lines exactly do? As far i can understand this tries to ignore invalid input in form of incorrect closing tags, but (apparently, huh) code already works that way?

Reply:
> What do these lines exactly do? As far i can understand this tries to ignore invalid input in form of incorrect closing tags, but (apparently, huh) code already works that way? The change below seems to be enough.

there seems to be a bug in the XMLParser in godot. For example, when the content of the XML file is
"<p>abc</p>\n", it will identify the node_type and node_name as:
type NodeType.NODE_ELEMENT name: p
type NodeType.NODE_TEXT
type NodeType.NODE_ELEMENT_END name: p
type NodeType.NODE_ELEMENT_END name: p        <---- This is a bug, the correct result should be NODE_TEXT
So I have to skip the wrong NODE_ELEMENT_END

Reply:
You are already skipping anything invalid past the last node (when queue is already empty), so there's no need for this "exception".